server.py

```diff
@@ -12,18 +20,66 @@ def start_server(self):
 
     def add_routes(self):
-        self.__server.add_url_rule('/shoes', '/shoes', self.get_shoes, methods=['GET'])
-        self.__server.add_url_rule('/create_shoes', '/create_shoes', self.create_shoes, methods=['POST'])
-        self.__server.add_url_rule('/remove_shoe', '/remove_shoe', self.remove_shoe, methods=['DELETE'])
-        self.__server.add_url_rule('/update_shoe', '/update_shoe', self.update_shoe, methods=['PATCH'])
+        self.__server.add_url_rule('/shoes', '/shoes', self.token_required(self.get_shoes), methods=['GET'])
+        self.__server.add_url_rule('/create_shoes', '/create_shoes', self.token_required(self.create_shoes), methods=['POST'])
+        self.__server.add_url_rule('/remove_shoe', '/remove_shoe', self.token_required(self.remove_shoe), methods=['DELETE'])
+        self.__server.add_url_rule('/update_shoe', '/update_shoe', self.token_required(self.update_shoe), methods=['PATCH'])
+        self.__server.add_url_rule('/user_signup', '/user_signup', self.user_signup, methods=['POST'])
+        self.__server.add_url_rule('/user_login', '/user_login', self.user_login,methods=['POST'])
 
     # @staticmethod
     # def __get_db_connection():
     #     conn = sqlite3.connect('shoes.db')
     #     conn.row_factory = sqlite3.Row
     #     return conn
+    secret_key = "random_string"
+    def token_required(self,f):
+        @wraps(f)
+        def decorated(*args, **kwargs):
+            token = None
+            if "Authorization" in request.headers:
+                token = request.headers["Authorization"]
+                print(token)
+            if not token:
+                return {
+                    "message": "Authentication Token is missing!",
+                    "data": None,
+                    "error": "Unauthorized"
+                }, 401
+            try:
+                data=jwt.decode(token, self.secret_key, algorithms=["HS256"])
+                # JSON string
+                json_string = '{"data": {"user_id": 1}, "exp": 1734884835}'
+
+                # Parse the string into a dictionary
+                parsed_json = json.loads(json_string)
+
+                # Access the value of 'user_id'
+                user_id = parsed_json['data']['user_id']
+                print(user_id)
+                query = ''' SELECT * from user_info where user_id = ? '''
+                params = (user_id,)
+                current_user=database.fetch_data(query,(user_id,))
+                if current_user is None:
+                    return {
+                    "message": "Invalid Authentication token!",
+                    "data": None,
+                    "error": "Unauthorized"
+                }, 401
+                
+            except Exception as e:
+                return {
+                    "message": "Something went wrong",
+                    "data": None,
+                    "error": str(e)
+                }, 500
+
+            return f(current_user, *args, **kwargs)
```

Comment:
is there any need to pass current_user here ? Bcoz it is not being used anywhere.